### ISSUE TITLE
Docker CLI  --password to --password-stdin

### ISSUE THREAD

**Issue by danielfischeroctopus** [OctopusDeploy/Issues#3802]
Currently our docker steps use the `--password flag`, docker 17.0.7 introduces `--password-stdin` 
```
Using STDIN prevents the password from ending up in the shell's history, or log-files.
```

This is considered more secure and docker will output the following warning to Octopus if it is not used.

```
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
```

Source: https://help.octopusdeploy.com/discussions/problems/58496
Related Docker docs: https://github.com/docker/cli/blob/master/docs/reference/commandline/login.md

**Comment by dsdenes** [comment]
`echo $(aws ecr get-authorization-token --output text --query 'authorizationData[].authorizationToken' | base64 -d | cut -d: -f2) | docker login -u AWS https://123456.dkr.ecr.eu-central-1.amazonaws.com --password-stdin`

**Comment by closedLoop** [comment]
We have used the following line in bash 

    export _DOCKER_REPO="$(aws ecr get-authorization-token --output text  --query 'authorizationData[].proxyEndpoint')"
    aws ecr get-login --no-include-email --region us-east-1 | awk '{print $6}' | docker login -u AWS --password-stdin $_DOCKER_REPO

**Comment by matt-richardson** [comment]
The challenge we have this one is that we have to support both older versions of docker as well as newer versions. This means we'll have to do some kind of version sniffing and use the appropriate parameters, which isn't fun. Doable, but not fun.

**Comment by catherinetcai** [comment]
@matt-richardson Not sure if this would be a great solution, but I rolled something like this in a Go lib I contribute to. I grabbed the Docker version output with `docker version --format {{.Client.Version}}`, which should just give you back the version in this format: `17.07.1-ce`.

Then I did a pretty gross regexp to just capture the major/minor (or I guess with the new release versions the year and month) and did a string comparison on them to decide whether or not to use the --password flag or the --password-stdin flag.
